checker/src/main/java/org/checkerframework/checker/nullness/NullnessNoInitVisitor.java

```diff
@@ -776,9 +788,38 @@ public void visitSwitchExpression17(Tree switchExprTree) {
             checkForNullability(
                     SwitchExpressionUtils.getExpression(switchExprTree), SWITCHING_NULLABLE);
         }
+        if (lintredundantNullComparison) {
+            ExpressionTree expression = SwitchExpressionUtils.getExpression(switchExprTree);
+            List<? extends CaseTree> cases = SwitchExpressionUtils.getCases(switchExprTree);
+            checkSwitchNullRedundant(expression, cases);
+        }
         super.visitSwitchExpression17(switchExprTree);
     }
 
+    /**
+     * Reports an error if the switch expression is @NonNull and the case expression is null
```

Comment:
```suggestion
     * Reports an error if the switch expression is {@code @NonNull} and the case expression is null
```

Reply:
It's not that the "case expression is null"... one of the case expressions in the case trees is null. Try to formulate it a bit clearer.